f8a_report/sentry_report_helper.py

```diff
@@ -0,0 +1,141 @@
+        """Retrieve results for selected worker from RDB."""
+        result = {}
+        try:
+            # Invoke Sentry API to run the error collection
+            auth = 'Bearer {token}'.format(token=self.sentry_token)
+            resp = requests.get(url=self.sentry_api_issues + '?statsPeriod=24h',
+                                headers={"Authorization": auth})
+            # Check for status code
+            # If status is not success, log it as an error
+            if resp.status_code == 200:
+                logger.info('Successfully invoked Sentry API \n {resp}'.format(resp=resp.json()))
+                # associate the retrieved data to result
+                result = self.normalize_sentry_data(start_date, end_date, resp.json())
+            else:
+                logger.error('Error received from Sentry API \n {resp}'.format(resp=resp.json()))
+        except requests.exceptions.RequestException as e:
+            logger.error('Unable to invoke Sentry API. Reason: %r' % e)
+        except requests.exceptions.Timeout as e:
+            logger.error('Timeout occured while invoking Sentry API. Reason: %r' % e)
+        return result
+
+    def normalize_sentry_data(self, start_date, end_date, errorlogs):
+        """Retrieve results for selected worker from RDB."""
+        report_type = 'sentry-error-data'
+        report_name = dt.strptime(end_date, '%Y-%m-%d').strftime('%Y-%m-%d')
+        result = {
+            "error_report": {}
+        }
+        # Iterating all the error logs
+        try:
+            for item in errorlogs:
+                errors = {}
+                events = self.retrieve_events(item['id'])
+                errors['id'] = item['id']
+                errors['last_seen'] = item['lastSeen']
+                errors[events['pods_impacted']] = item['metadata']['type'] + ": " + \
+                    item['metadata']['value'] if item['metadata'].get('type')\
+                    else item['metadata']['title']
+                errors['stacktrace'] = events['stacktrace']
+                # Detecting the endpoint services
+                server_name = "-".join(events['pods_impacted'].split("-")[:-2])
+                result['error_report'][server_name] = result['error_report'][server_name] \
+                    if result['error_report'].get(server_name) else {}
+                # Calculating total errors
+                result['error_report'][server_name]['total_errors'] = \
+                    result.get('error_report').get(server_name).get('total_errors', 0) + 1
+                if not result['error_report'][server_name].get('errors'):
+                    result['error_report'][server_name]['errors'] = []
+                result['error_report'][server_name]['errors'].append(errors)
+        except KeyError as e:
+            logger.error('Key not found while parsing. Reason: %r' % e)
+            # Saving the final report in the relevant S3 bucket
```

Comment:
@animuk @miteshvp are we fine with storing a blank json block as error_report on S3 if a KeyError is encountered? I do not have any strong opinion about any of the options.

Reply:
@samuzzal-choudhury let it be that way. There will not be any report if we handle this. We will revisit in future if there is any need for this

Reply:
@animuk fine. cc: @miteshvp